jekyll/_cci2/env-vars.md

```diff
@@ -16,19 +16,40 @@ This document describes using environment variables in CircleCI in the following
 
 ## Overview
 
-Environment variables are used according to a specific precedence order, as follows:
-
-1. Environment variables declared inside a shell command in a `run` step, for example `FOO=bar make install`.
-2. Environment variables declared with the `environment` key for a `run` step.
-3. Environment variables set with the `environment` key for a job.
-4. Environment variables set with the `environment` key for a container.
-5. Context environment variables (assuming the user has access to the Context). See the [Contexts]( {{ site.baseurl }}/2.0/contexts/) documentation for instructions.
-6. Project-level environment variables set on the Project Settings page.
-7. Special CircleCI environment variables defined in the [CircleCI Environment Variable Descriptions]({{ site.baseurl }}/2.0/env-vars/#circleci-built-in-environment-variables) section of this document.
-
-Environment variables declared inside a shell command `run step`, for example `FOO=bar make install`, will override environment variables declared with the `environment` and `contexts` keys. Environment variables added on the Contexts page will take precedence over variables added on the Project Settings page. Finally, special CircleCI environment variables are loaded.
-
-## Adding Project-Level Environment Variables
+Environment variables can be set at several scope levels.
+From general to specific, these levels are:
+
+1. Environment variables [built into CircleCI](#circleci-built-in-environment-variables).
+Automatically available.
+2. [Project-level](#setting-project-level-environment-variables) environment variables.
+Set within the CircleCI application.
+3. [Context-level]({{ site.baseurl }}/2.0/contexts/) environment variables.
+Set within the CircleCI application.
+4. [Job-level](#setting-job-level-environment-variables) environment variables.
+Set in your `.circleci/config.yml` file.
+5. [Container-level](#setting-container-level-environment-variables) environment variables.
+Set in your `.circleci/config.yml` file.
+6. [Step-level](#setting-step-level-environment-variables) environment variables.
+Set in your `.circleci/config.yml` file.
+7. Environment variables [set within shell commands](#setting-command-level-environment-variables).
+Set in your `.circleci/config.yml` file.
+
+CircleCI also loads environment variables in this order.
```

Comment:
No, they are not used in the order from lowest level to highest as you have it here. This is what we determined in the March revision and clarified.

Reply:
@marcomorain reviewed this new PR for technical accuracy and approved it. There _might_ be different behavior for the special, built-in environment variables, but the rest of this load order (general to specific) is correct. Environment variables at the configuration level will overwrite variables with the same name at a higher level.

Reply:
Where did you get this information though? I think this is not what we agreed before when this came up in customer feedback and you are changing it now, but I'm not sure why.